### ISSUE TITLE
On QCQP in Gurobi v9.0

### ISSUE THREAD

**Issue by ccoffrin** [JuliaOpt/Gurobi.jl#309]
I noticed the following,
- `QCPDual` seems to be set to 1 by default, which differs from Gurobi's documented semantics
- When `QCPDual` is not set, the SolveTime attribute seems to return only the dual solve time and not the overall solve time for primal and dual.

**Comment by odow** [comment]
We set `QCPDual` because too many people were asking why duals weren't returned when they solved a QP.

The second point might need to be followed up with Gurobi directly?

**Comment by odow** [comment]
Is there anything actionable here?

**Comment by ccoffrin** [comment]
I would cast a vote for this wrapper not deviating Gurobi's default settings.  Users should know how the solver they use works.

I have not had a chance to contact Gurobi the time reporting issue but I plan to.

**Comment by mlubin** [comment]
As long as the support burden falls on us, I think it's reasonable to choose defaults that reduce that burden. It would be nicer, however, if we can keep Gurobi's default settings and produce useful warnings to point people to set `QCPDual=1` if they need duals.

**Comment by ccoffrin** [comment]
If possible, I think a helpful warning message would be a great solution.

**Comment by odow** [comment]
It seems like at some point Gurobi changed their error message so that it now prints said warning:
```Julia
julia> model = Model(() -> Gurobi.Optimizer(QCPDual = 0))
Academic license - for non-commercial use only
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: Gurobi

julia> @variable(model, x >= 0)
x

julia> @constraint(model, x^2 >= 2)
x² ≥ 2.0

julia> @objective(model, Min, x)
x

julia> optimize!(model)
Academic license - for non-commercial use only
Optimize a model with 0 rows, 1 columns and 0 nonzeros
Model has 1 quadratic constraint
Coefficient statistics:
  Matrix range     [0e+00, 0e+00]
  QMatrix range    [1e+00, 1e+00]
  Objective range  [1e+00, 1e+00]
  Bounds range     [0e+00, 0e+00]
  RHS range        [0e+00, 0e+00]
  QRHS range       [2e+00, 2e+00]
Presolve removed 0 rows and 1 columns
Presolve time: 0.00s
Presolve: All rows and columns removed

Barrier solved model in 0 iterations and 0.00 seconds
Optimal objective 1.41421356e+00

julia> has_duals(model)
false

julia> dual(LowerBoundRef(x))
Warning: to get QCP duals, please set parameter QCPDual to 1

ERROR: Gurobi.GurobiError(10005, "Unable to retrieve attribute 'RC'")
Stacktrace:
 [1] get_dblattrelement(::Gurobi.Model, ::String, ::Int64) at /Users/oscar/.julia/packages/Gurobi/GeYlA/src/grb_attrs.jl:68
```

**Comment by ccoffrin** [comment]
For anyone who might come across this issue.  Concerning this point, "When QCPDual=1, the SolveTime attribute seems to return only the dual solve time and not the overall solve time for primal and dual."  Upon further inspection this does not appear to be the case.  Runtime is addative across the primal and dual solve.
